packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -70,6 +73,20 @@ export type QueuedRequestControllerOptions = {
   messenger: QueuedRequestControllerMessenger;
 };
 
+/**
+ * A queued request.
+ */
+type QueuedRequest = {
+  /**
+   * The origin of the queued request.
+   */
+  origin: string;
+  /**
+   * A callback used to continue processing the request, called when the request is dequeued.
+   */
+  processRequest: (error: unknown) => void;
+};
+
 /**
  * Controller for request queueing. The QueuedRequestController manages the orderly execution of enqueued requests
```

Comment:
Nit: This should be updated to reflect the new behavior and explain its purpose more effectively

Reply:
Done in https://github.com/MetaMask/core/pull/3781/commits/aab7c31d982aa599801d13d7d880b5f8b6bfad9a